logcli: add support for bearer token authentication (#3585)
grafana/loki: docs/sources/getting-started/logcli.md

A command-line for loki.

Flags:
      --help                  Show context-sensitive help (also try --help-long and --help-man).
      --version               Show application version.
  -q, --quiet                 Suppress query metadata.
      --stats                 Show query statistics.

>   -o, --output=default        Specify output mode [default, raw, jsonl]. raw suppresses log labels and timestamp.

Author: @achatterjee-grafana I don't understand the sense behind these improvements in the `getting-started/logcli.md`. This text is from the [source code](https://github.com/grafana/loki/blob/0a02aecdb993da9d56714d756a6be73039e5317d/cmd/logcli/main.go#L26), what is the point of changing it here?

  series [<flags>] <matcher>
    Run series query.

>     The "series" command will take the provided label matcher and return all the log streams found in the time window.

Reviewer: ```suggestion
    The "series" command takes the label matcher (provided) and returns all the log streams found in the time window.
```

The output of the log can be specified with the "-o" flag, for example, "-o raw" for the raw output format.

> The "query" command will output extra information about the query and its results, such as the API URL, set of common labels, and set of excluded labels. This extra information can be suppressed with the --quiet flag.

Reviewer: ```suggestion
The "query" command returns extra information about the query and its results, such as the API URL, a set of common labels, and a set of excluded labels. This extra information can be hidden with the --quiet flag.
```

      --stats                 Show query statistics
  -o, --output=default        Specify output mode [default, raw, jsonl]. raw suppresses log labels and timestamp.
  -z, --timezone=Local        Specify the timezone to use when formatting output timestamps [Local, UTC]
      --cpuprofile=""         Specify the location for writing a CPU profile.
      --memprofile=""         Specify the location for writing a memory profile.
      --addr="http://localhost:3100"
                              Server address. Can also be set using LOKI_ADDR env var.
      --username=""           Username for HTTP basic auth. Can also be set using LOKI_USERNAME env var.
      --password=""           Password for HTTP basic auth. Can also be set using LOKI_PASSWORD env var.
      --ca-cert=""            Path to the server Certificate Authority. Can also be set using LOKI_CA_CERT_PATH env var.
      --tls-skip-verify       Server certificate TLS skip verify.
      --cert=""               Path to the client certificate. Can also be set using LOKI_CLIENT_CERT_PATH env var.
      --key=""                Path to the client certificate key. Can also be set using LOKI_CLIENT_KEY_PATH env var.

>       --org-id=""             adds X-Scope-OrgID to API requests for representing tenant ID. Useful for requesting tenant data when bypassing an auth gateway.

Reviewer: ```suggestion
      --org-id=""             Add X-Scope-OrgID to API requests for representing tenant ID. Useful for requesting tenant data when bypassing an auth gateway.
```

      --bearer-token=""       adds the Authorization header Bearer to API requests for authentication purposes. Can also be set using LOKI_BEARER_TOKEN env var.

>       --bearer-token-file=""  adds the Authorization header Bearer to API requests for authentication purposes. Can also be set using LOKI_BEARER_TOKEN_FILE env var.

Reviewer: ```suggestion
      --bearer-token-file=""  Add the Authorization header Bearer to API requests for authentication purposes. Can also be set using LOKI_BEARER_TOKEN_FILE env var.
```

      --since=1h              Lookback window.
      --from=FROM             Start looking for logs at this absolute time (inclusive)
      --to=TO                 Stop looking for logs at this absolute time (exclusive)

>       --analyze-labels        Printout a summary of labels including count of label value combinations, useful for debugging high cardinality series

Reviewer: ```suggestion
      --analyze-labels        Prints out a summary of labels including the count of label value combinations, useful for debugging high cardinality series
```
